packages/language-server/src/diagnostics/DiagnosticProvider.ts

```diff
@@ -40,16 +72,78 @@ export class DiagnosticProvider {
             return createDiagnostic(node, `Empty ${node.type} block`);
         }
 
+
+        if (isPathInsideTemplateEmbedding(node)) {
+            const path = getStringNodeValue(node);
+            const document = await this.documentCache.resolveByTwigPath(path);
+
+            if (!document) {
+                return createDiagnostic(node, `Template "${path}" not found`, DiagnosticSeverity.Error)
+            }
+
+            // found template, no errors
+            return null;
+        }
+
+        if (isBlockIdentifier(node)) {
+            if (node.parent?.type !== 'block') {
+                let extendedDocument: Document | undefined = document;
+                let documentName = "this document";
+                
+                const blockArgumentNode = node.parent!.namedChildren[0];
+                const templateArgumentNode = node.parent!.namedChildren[1];
+
+                if (templateArgumentNode) {
+                    const path = getStringNodeValue(templateArgumentNode);
+                    const document = await this.documentCache.resolveByTwigPath(path);
+    
+                    if (node.equals(templateArgumentNode)) {
+                        if (!document) {
+                            return createDiagnostic(templateArgumentNode, `Template "${path}" not found`, DiagnosticSeverity.Error)
+                        }
+                        // found template, no errors
+                        // block existence will be checked in next pass
+                        return null;
+                    }
+
+                    extendedDocument = document;
+                    documentName = path;
+                }
+
+                const blockName = blockArgumentNode.type === 'string'
+                    ? getStringNodeValue(blockArgumentNode)
+                    : blockArgumentNode.text;
+
+                let found = false;
+                while (extendedDocument) {
+                    const blockSymbol = extendedDocument.getBlock(blockName);
+                    if (!blockSymbol || positionsEqual(blockSymbol.nameRange.start, getNodeRange(blockArgumentNode).start)) {
+                        if (!extendedDocument.locals.extends) {
+                            extendedDocument = void 0;
+                        } else {
+                            extendedDocument = await this.documentCache.resolveByTwigPath(extendedDocument.locals.extends);
+                        }
+                        continue;
+                    }
+                    found = true;
+                    break;
+                }
+                if (!found) {
+                    return createDiagnostic(blockArgumentNode, `Block "${blockName}" not found in ${documentName}`, DiagnosticSeverity.Error)
+                }
+            }
+        }
+
         return null;
     }
 
-    validateTree(tree: Tree) {
+    async validateTree(tree: Tree, document: Document) {
```

Comment:
With this change of signature, the `diagnostics.test.ts` is broken

Also, I'd rather define it as `validateTree(document: Document)` and add a `const { tree } = document;`